### PR TITLE
Pretty frontend + Regenerate build files

### PR DESCRIPTION
**Pull request by akgupta89**
**Summary**
Made the website responsive using bootstrap components and made docker-compose rebuild the production web page. You have to manually run "npm run build" to regenerate the production website normally.

**Preview**
Screen Shot 2019-04-03 at 8 27 30 PM

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -5,10 +5,11 @@ services:
     build:
       dockerfile: Dockerfile
       context: .
-    ports:
-      - "5000:5000"
+    command: ["npm", "run", "build"]
```
Comment (youngj) [inline docker-compose.yml]: This creates an optimized production build and exits without running the flask backend.

I think we shouldn't create an optimized production build here because:
- it's slow, which adds extra time waiting for your development environment to be ready
- it's usually not needed because people will be making changes to the front end at http://localhost:3000 which has live reloading 
- it's likely to confuse developers who look at http://localhost:5000 , try to make changes to the front end code and get confused because none of their changes are having any effect.

I'd like to keep the previous functionality and not create an optimized production build when running `docker-compose up`.

File: frontend/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,17372 @@
+{
```
Comment (youngj) [inline frontend/package-lock.json]: package-lock.json should be deleted -- we are already using yarn.lock and `yarn build` / `yarn install`, and shouldn't run npm commands that generate package-lock.json (unless we make a decision to stop using yarn, but if we do that should be in a separate PR that removes all instances of yarn).

File: frontend/src/App.jsx [hunk 3]
```diff
@@ -23,18 +23,8 @@ class App extends Component {
   render() {
     const { graphData, graphError, routes } = this.props;
     return (
-      <div className={css`
-        display: grid;
```
Comment (youngj) [inline frontend/src/App.jsx]: This removes the grid layout that was added last week in https://github.com/trynmaps/metrics-mvp/pull/45 . There are probably differing ideas about what CSS layout we're heading towards. The one column layout adapts well to mobile, though, which is nice. An advantage of using a grid is that it makes more efficient use of the screen space when the screen is wider (compared to using the entire screen width for the control panel). We could use flex box as a more responsive alternative to a grid layout: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Relationship_of_Grid_Layout

File: frontend/package-lock.json (hunk 2 above)
Comment (sravanti) [reply frontend/package-lock.json]: Ah I thought we had decided on npm and not yarn—@youngj is that not correct? In any case, agreed that we just need to use one package manager.

File: frontend/src/App.jsx (hunk 3 above)
Comment (sravanti) [reply frontend/src/App.jsx]: +1 here—with CSS grid, we can specify a one-column layout specifically for mobile-width screens, so let's investigate that first before removing CSS grid.

File: frontend/src/components/ControlPanel.jsx [hunk 4]
```diff
@@ -113,43 +113,42 @@ class ControlPanel extends Component {
...
-          grid-column: col1-start / col3-start;
-           grid-row: row2-start ;
-      `
-      }
-        >
-          <DatePicker value={date} onChange={this.setDate} />
-          <DropdownControl title="Route" name='route' value={routeId}
+      <div className="row">
+        <div className="col-12 form-group"><DatePicker value={date} onChange={this.setDate} className={css`
+          display: block !important;
```
Comment (sravanti) [inline frontend/src/components/ControlPanel.jsx]: Is there a reason why you're using `!important` here? It's typically best practice to avoid using it since it does disrupt the CSS specificity rules and should only be used in extreme edge cases (see this article: https://j11y.io/css/dont-use-important/).

File: docker-compose.yml (hunk 1 above)
Comment (hathix) [reply docker-compose.yml]: Agree; let's not do an optimized production build here.

**Comment by akgupta89** [comment]
I think @sravanti made my PR obsolete, but we created two tasks from this:

1. yarn vs npm
2. update the documentation and provide the command to regenerate production files

**Comment by hathix** [comment]
Have made Trello cards for these. Thanks Arun!
